src/components/MapView/DateSelector/index.tsx

```diff
@@ -43,38 +51,148 @@ function DateSelector({ availableDates = [], classes }: DateSelectorProps) {
   const dispatch = useDispatch();
   const { startDate: stateStartDate } = useSelector(dateRangeSelector);
 
-  const stateStartDateYear = moment(stateStartDate).year();
   const [selectedDate, setSelectedDate] = useState(moment(stateStartDate));
-  const selectedMonth = selectedDate.month();
+  const [dateRange, setDateRange] = useState([
+    {
+      value: 0,
+      label: '',
+      month: '',
+      isFirstday: false,
+    },
+  ]);
+
+  const [timelinePosition, setTimelinePosition] = useState({ x: 0, y: 0 });
+  const [pointerPosition, setPointerPosition] = useState({ x: 0, y: 0 });
+
+  const dateRef = useRef(availableDates);
 
-  const selectedYear = selectedDate.year();
+  const timeLine = useRef(null);
+  const timeLineWidth = get(timeLine.current, 'offsetWidth', 0);
 
+  // Move the slider automatically so that the pointer always visible
   useEffect(() => {
+    let x = 0;
+    if (pointerPosition.x >= dateRange.length * 10 - timeLineWidth) {
+      // eslint-disable-next-line fp/no-mutation
+      x = timeLineWidth - dateRange.length * 10;
+    } else if (pointerPosition.x > timeLineWidth) {
+      // eslint-disable-next-line fp/no-mutation
+      x = -1 * pointerPosition.x + timeLineWidth / 2;
+    }
+    if (
+      -1 * timelinePosition.x > pointerPosition.x ||
+      -1 * timelinePosition.x + timeLineWidth < pointerPosition.x
+    ) {
+      setTimelinePosition({ x, y: 0 });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
We should avoid ignoring this eslint rule, since it helps make sure we use `useEffect` as it was meant to be used, keeping its behavior predictable. I recommend [this answer](https://stackoverflow.com/a/58959298/5279269) to help gain a grasp why, it helped for me. Does the code work without this ignore?

Reply:
It works if i include all the dependencies and remove the ignore. is that alright?